api/v01/entities/vrp_input.rb

```diff
@@ -0,0 +1,521 @@
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Mapotempo. If not, see:
+# <http://www.gnu.org/licenses/agpl.html>
+#
+require './api/v01/vrp'
+
+module VrpInput
+  extend Grape::API::Helpers
+
+  params :input do
+    optional(:vrp, type: Hash, documentation: { param_type: 'body' }, coerce_with: ->(c) { c.has_key?('filename') ? JSON.parse(c[:tempfile].read) : c }) do
+      use :request_object
+    end
+    use :request_files
+    exactly_one_of :vrp, :vehicles # either it is a json (and :vrp is required) or it is a csv (and :vehicles is required)
+  end
+
+  # Input as expected in JSON
+  params :request_object do
+    optional(:name, type: String, desc: 'Name of the problem, used as tag for all element in order to name plan when importing returned .csv file')
+    optional(:matrices, type: Array, documentation: { desc: 'Define all the distances between each point of problem' }) do use :vrp_request_matrices end
+
+    optional(:points, type: Array, documentation: { desc: 'Particular place in the map' }) do use :vrp_request_point end
+
+    optional(:units, type: Array, documentation: { desc: 'The name of a Capacity/Quantity' }) do use :vrp_request_unit end
+
+    optional(:rests, type: Array, documentation: { desc: 'Break within a vehicle tour' }) do use :vrp_request_rest end
+
+    optional(:zones, type: Array, desc: '') do use :vrp_request_zone end
+
+    requires(:vehicles, type: Array, documentation: { desc: 'Usually represent a work day of a particular driver/vehicle' }) do
+      use :vrp_request_vehicle
+    end
+
+    optional(:services, type: Array, allow_blank: false, documentation: { desc: 'Independent activity, which does not require a context' }) do
+      use :vrp_request_service
+    end
+    optional(:shipments, type: Array, allow_blank: false, documentation: { desc: 'Link directly one activity of collection to another of drop off. Not available with periodic heuristic.' }) do
+      use :vrp_request_shipment
+    end
+    at_least_one_of :services, :shipments
+
+    optional(:relations, type: Array, desc: 'Not available with periodic heuristic') do use :vrp_request_relation end
+
+    optional(:subtours, type: Array, desc: 'Not available with periodic heuristic') do use :vrp_request_subtour end
+
+    optional(:routes, type: Array, desc: '') do use :vrp_request_route end
+
+    optional(:configuration, type: Hash, documentation: { desc: 'Describe the limitations of the solve in term of computation' }) do
+      use :vrp_request_configuration
+    end
+  end
+
+  # Input as expected in CSV
+  params :request_files do # rubocop:disable Metrics/BlockLength
+                           # We expect here to keep the definition of the high level model in a single place
+    optional :points, type: Array,
+                      documentation: { hidden: true, desc: 'Warning : CSV Format expected here ! Particular place in the map' },
```

Comment:
Good practice suggest that there should be a dot if there are at least two sentences in the description.
```suggestion
                      documentation: { hidden: true, desc: 'Warning : CSV Format expected here ! Particular place in the map.' },
```